sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollingStrategy.java

```diff
@@ -0,0 +1,101 @@
+     * the long running operation will be terminated and none of the other methods will be invoked.
+     *
+     * @param response the response from the initial method call to activate the long running operation
+     * @param pollingContext the {@link PollingContext} for the current polling operation
+     * @param pollResponseType the {@link TypeReference} of the response type from a polling call, or BinaryData if raw
+     *                         response body should be kept. This should match the generic parameter {@link U}.
+     * @return a publisher emitting the long running operation status
+     */
+    Mono<LongRunningOperationStatus> onInitialResponse(Response<?> response, PollingContext<T> pollingContext,
+                                            TypeReference<T> pollResponseType);
+
+    /**
+     * Parses the response from the polling URL into a {@link PollResponse}, and stores information
+     * useful for further polling and final response in the {@link PollingContext}. The result must have the
+     * {@link LongRunningOperationStatus} specified, and the entire polling response content as a {@link BinaryData}.
+     *
+     * @param pollingContext the {@link PollingContext} for the current polling operation
+     * @param pollResponseType the {@link TypeReference} of the response type from a polling call, or BinaryData if raw
+     *                         response body should be kept. This should match the generic parameter {@link U}.
+     * @return a publisher emitting the a poll response containing the status and the response content
+     */
+    Mono<PollResponse<T>> poll(PollingContext<T> pollingContext, TypeReference<T> pollResponseType);
+
+    /**
+     * Parses the response from the final GET call into the result type of the long running operation.
+     *
+     * @param pollingContext the {@link PollingContext} for the current polling operation
+     * @param resultType the {@link TypeReference} of the final result object to deserialize into, or BinaryData if
+     *                   raw response body should be kept.
+     * @return a publisher emitting the final result
+     */
+    Mono<U> getResult(PollingContext<T> pollingContext, TypeReference<U> resultType);
+
+    /**
+     * Cancels the long running operation if service supports cancellation. If service does not support cancellation
+     * then the implementer should return Mono.error with an error message indicating absence of cancellation.
+     *
+     * @param pollingContext the {@link PollingContext} for the current polling operation, or null if the polling has
+     *                       started in a {@link SyncPoller}
+     * @param initialResponse the response from the initial operation
+     * @return a publisher emitting the cancellation response content
+     */
+    Mono<T> cancel(PollingContext<T> pollingContext, PollResponse<T> initialResponse);
```

Comment:
All implementation returns an error.
```
    @Override
    public Mono<T> cancel(PollingContext<T> pollingContext, PollResponse<T> initialResponse) {
        return Mono.error(new IllegalStateException("Cancellation is not supported."));
    }
```

Reply:
Using a default implementation in `PollingStrategy` interface.